docs/how-to/internal/release.rst

```diff
@@ -0,0 +1,72 @@
+============================
+How to cut a Rubicon release
+============================
+
+The release infrastructure for Rubicon is semi-automated, using Github
+Actions to formally publish releases.
+
+This guide assumes that you have an ``upstream`` remote configured on your
+local clone of the Rubicon repository, pointing at the official repository.
+It also assumes that you have a local ``release`` branch that tracks
+``upstream/master``.
+
+You should have a checkout of a personal fork of the Rubicon repository; to
+configure this fork for a release, run::
+
+    $ git remote add upstream git@github.com:beeware/rubicon-objc.git
+    $ git clone upstream/master release
+
+The procedure for cutting a new release is as follows:
+
+1. Refresh your release branch::
+
+    $ git checkout release
+    $ git fetch upstream
+    $ git pull
+
+   Check that the HEAD of release now matches upstream/master.
```

Comment:
Possibly? I'll need to try and see... what I've documented here is the process I've fallen into because messing with git remotes always leaves me feeling like I've participated in some sort of black magick; this approach it worked, so I've stuck with it. 

(To be clear, I'm clear on the *concepts* behind git; I just find the UX around git's CLI gets really messy and confused about how it sometimes expects remote/branch, and sometimes the remote is assumed, except when it isn't and you have to specify the remote as a standalone argument without a prefix on the branch, except when its a tag... and so on).

Reply:
To be clear, a detached checkout like this *won't* work if you need to commit something (version bump, towncrier) on top of the current state of `upstream/master` - for that you need a proper remote-tracking branch that you can push. (Well, knowing Git you probably don't *need* one, but you *should* be using one, because otherwise things get messy.) This only works if you want to create a tag *exactly* at the state of `upstream/master`. Although in that case you could simply do `git tag v1.2.3 upstream/master`...

Yeah, Git's CLI usability isn't always great. Things are improving though - for example they recently added [`git switch`](https://git-scm.com/docs/git-switch) and [`git restore`](https://git-scm.com/docs/git-restore), which are intended as replacements for `git checkout` and (partially) `git branch` with more usable CLI syntax.